modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIPropertyRelevantObjectEntryModelListener.java

```diff
@@ -43,15 +46,54 @@ public String getObjectDefinitionExternalReferenceCode() {
 	public void onBeforeCreate(ObjectEntry objectEntry)
 		throws ModelListenerException {
 
-		_validate(objectEntry);
+		_validate(objectEntry.getValues());
 	}
 
 	@Override
 	public void onBeforeUpdate(
 			ObjectEntry originalObjectEntry, ObjectEntry objectEntry)
 		throws ModelListenerException {
 
-		_validate(objectEntry);
+		Map<String, Serializable> values = objectEntry.getValues();
+
+		long apiSchemaId = (long)values.get(
+			"r_apiSchemaToAPIProperties_c_apiSchemaId");
+
+		if (apiSchemaId != 0) {
```

Comment:
You could use [this method](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/portal-kernel/src/com/liferay/portal/kernel/util/MapUtil.java#L162) directly:

```
long apiSchemaId = MapUtil.getLong(
	values, "r_apiSchemaToAPIProperties_c_apiSchemaId")
```

Reply:
Noted for future PRs, thanks @ccorreagg !!